src/runtime_src/hip/api/hipMemory.cpp

```diff
@@ -0,0 +1,222 @@
+static void
+hipHostMalloc(void** ptr, size_t size, unsigned int flags)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Free memory allocated by the hipHostMalloc().
+static void 
+hipHostFree(void* ptr)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Free memory allocated by the hipMalloc(). 
+static void
+hipFree(void* ptr)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Register host memory so it can be accessed from the current device.
+static void
+hipHostRegister(void* hostPtr,
+                size_t sizeBytes,
+                unsigned int flags)
+{
+    throw std::runtime_error("Not implemented");   
+}
+
+// Un-register host pointer.
+static void
+hipHostUnregister(void* hostPtr)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Get Device pointer from Host Pointer allocated through hipHostMalloc().
+static void
+hipHostGetDevicePointer(void** devPtr,
+                        void* hstPtr,
+                        unsigned int flags)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+// Copy data from src to dst. 
+static void
+hipMemCpy(void* dst,
+          const void* src,
+          size_t sizeBytes,
+          hipMemcpyKind kind)
+{
+    throw std::runtime_error("Not implemented");    
+}
+
+} // xrt::core::hip
+
+// Allocate memory on the device.
+hipError_t
+hipMalloc(void** ptr, size_t size)
+{
+    *ptr = nullptr;
+    if (size == 0)
+    {
+        return hipSuccess;
+    }
+    try 
+    {
+        xrt::core::hip::hipMalloc(ptr, size);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Allocates device accessible host memory.
+hipError_t 
+hipHostMalloc(void** ptr, size_t	size, unsigned int flags)
+{
+    *ptr = nullptr;
+    if (size == 0)
+    {
+        return hipSuccess;
+    }
+    try 
+    {
+        xrt::core::hip::hipHostMalloc(ptr, size, flags);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Free memory allocated by the hipHostMalloc().
+hipError_t
+hipHostFree(void* ptr)
+{
+    try 
+    {
+        xrt::core::hip::hipHostFree(ptr);
+        return hipSuccess;
+    }
+    catch 
+    (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Free memory allocated by the hipMalloc().
+hipError_t
+hipFree(void* ptr)
+{
+    try 
+    {
+        xrt::core::hip::hipFree(ptr);
+        return hipSuccess;
+    }
+    catch (const std::exception& ex) 
+    {
+        xrt_core::send_exception_message(ex.what());
+    }
+    return hipErrorUnknown;
+}
+
+// Register host memory so it can be accessed from the current device.
+hipError_t
+hipHostRegister(void* hostPtr,
+    size_t sizeBytes,
+    unsigned int flags)
+{
```

Comment:
Fix the indentation to match `hipHostGetDevicePointer`.

Reply:
Done.